go/vt/schemamanager/local_controller.go

```diff
@@ -0,0 +1,222 @@
+// Copyright 2015, Google Inc. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package schemamanager
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+	"time"
+
+	log "github.com/golang/glog"
+)
+
+// LocalController listens to the specified schema change dir and applies schema changes.
+// schema change dir lay out
+//            |
+//            |----keyspace_01
+//                 |----input
+//                      |---- create_test_table.sql
+//                      |---- alter_test_table_02.sql
+//                      |---- ...
+//                 |----complete // contains completed schema changes in yyyy/MM/dd
+//                      |----2015
+//                           |----01
+//                                |----01
+//                                     |--- create_table_table_02.sql
+//                 |----log // contains detailed execution information about schema changes
+//                      |----2015
+//                           |----01
+//                                |----01
+//                                     |--- create_table_table_02.sql
+//                 |----error // contains failed schema changes
+//                      |----2015
+//                           |----01
+//                                |----01
+//                                     |--- create_table_table_03.sql
+// Schema Change Files: ${keyspace}/input/*.sql
+// Error Files:         ${keysapce}/error/${YYYY}/${MM}/${DD}/*.sql
+// Log Files:           ${keysapce}/log/${YYYY}/${MM}/${DD}/*.sql
+// Complete Files:      ${keysapce}/compelte/${YYYY}/${MM}/${DD}/*.sql
+type LocalController struct {
+	schemaChangeDir string
+	keyspace        string
+	sqlPath         string
+	sqlFilename     string
+	errorDir        string
+	logDir          string
+	completeDir     string
+}
+
+// NewLocalController creates a new LocalController instance.
+func NewLocalController(schemaChangeDir string) *LocalController {
+	return &LocalController{
+		schemaChangeDir: schemaChangeDir,
+	}
+}
+
+// Open goes through the schema change dir and find a keyspace with a pending
+// schema change.
+func (controller *LocalController) Open() error {
+	// find all keyspace directories.
+	fileInfos, err := ioutil.ReadDir(controller.schemaChangeDir)
+	if err != nil {
+		return err
+	}
+	for _, fileinfo := range fileInfos {
```

Comment:
it is intended, only one schema change will be executed at a time. All remaining changes will be picked up by the next runs.